modules/core/src/shaderlib/lighting/lighting.js

```diff
@@ -18,107 +18,79 @@
-function getUniforms(opts = INITIAL_MODULE_OPTIONS) {
-  if (!opts.lightSettings) {
-    return {};
+  if (lightEffect.ambientLight) {
+    lightSourceUniforms['lighting_ambientLight.color'] = lightEffect.ambientLight.color;
+    lightSourceUniforms['lighting_ambientLight.intensity'] = lightEffect.ambientLight.intensity;
   }
 
-  const {
-    coordinateSystem,
-    coordinateOrigin,
-    lightSettings: {
-      numberOfLights = 1,
-
-      lightsPosition = DEFAULT_LIGHTS_POSITION,
-      lightsStrength = DEFAULT_LIGHTS_STRENGTH,
-      coordinateSystem: fromCoordinateSystem,
-      coordinateOrigin: fromCoordinateOrigin,
-      modelMatrix = null,
-
-      ambientRatio = DEFAULT_AMBIENT_RATIO,
-      diffuseRatio = DEFAULT_DIFFUSE_RATIO,
-      specularRatio = DEFAULT_SPECULAR_RATIO
-    }
-  } = opts;
+  let index = 0;
+  for (const i in lightEffect.pointLights) {
+    const pointLight = lightEffect.pointLights[i];
+    lightSourceUniforms[`lighting_pointLight[${index}].color`] = pointLight.color;
+    lightSourceUniforms[`lighting_pointLight[${index}].intensity`] = pointLight.intensity;
+    lightSourceUniforms[`lighting_pointLight[${index}].position`] = pointLight.position;
+    index++;
+  }
+  lightSourceUniforms.lighting_pointLightNumber = index;
 
-  // Pre-project light positions
-  const lightsPositionWorld = getMemoizedLightPositions({
-    lightsPosition,
-    numberOfLights,
-    viewport: opts.viewport,
-    modelMatrix,
-    coordinateSystem,
-    coordinateOrigin,
-    fromCoordinateSystem,
-    fromCoordinateOrigin
-  });
+  for (const i in lightEffect.directionalLights) {
+    const directionalLight = lightEffect.directionalLights[i];
+    lightSourceUniforms[`lighting_directionalLight[${index}].color`] = directionalLight.color;
+    lightSourceUniforms[`lighting_directionalLight[${index}].intensity`] =
+      directionalLight.intensity;
+    lightSourceUniforms[`lighting_directionalLight[${index}].direction`] =
+      directionalLight.direction;
+    index++;
+  }
+  lightSourceUniforms.lighting_directionalLightNumber = lightEffect.directionalLights.length;
 
-  return {
-    lighting_lightPositions: lightsPositionWorld,
-    lighting_lightStrengths: lightsStrength,
-    lighting_ambientRatio: ambientRatio,
-    lighting_diffuseRatio: diffuseRatio,
-    lighting_specularRatio: specularRatio,
-    lighting_numberOfLights: numberOfLights
-  };
+  return lightSourceUniforms;
 }
 
-// Pre-project light positions
-function preprojectLightPositions({
-  lightsPosition,
-  numberOfLights,
-  viewport,
-  modelMatrix,
-  coordinateSystem,
-  coordinateOrigin,
-  fromCoordinateSystem,
-  fromCoordinateOrigin
-}) {
-  const projectionParameters = {
-    viewport,
-    modelMatrix,
-    coordinateSystem,
-    coordinateOrigin,
-    fromCoordinateSystem,
-    fromCoordinateOrigin
-  };
+function getMaterialUniforms(material) {
+  const materialUniforms = {};
+  materialUniforms.lighting_ambient = material.ambient;
```

Comment:
IS the ambient really a property of the material? I kind of think of it as a light source, rather than a material property?

Reply:
Here is the ambient ratio property in material side, for phone model, each material need (ambient, diffuse and specular) ratios.